src/core/ext/transport/chttp2/transport/context_list.h

```diff
@@ -32,18 +31,46 @@ namespace grpc_core {
 class ContextList {
  public:
   // Creates a new element with \a context as the value and appends it to the
-  // list.
-  static void Append(ContextList** head, grpc_chttp2_stream* s);
+  // list. Each RPC/stream is associated with a unique \a context. This method
+  // is invoked when a chunk of data stored in an outgoing buffer is going to be
+  // sent over the wire. A data chunk being written over the wire is multiplexed
+  // with bytes from multiple RPCs. If one such RPC is traced, we store the
+  // following information about the traced RPC:
+  // - byte_offset_in_stream: Number of bytes belonging to that traced RPC which
+  // have been sent so far from the start of the RPC stream.
+  // - relative_start_pos_in_chunk: Starting offset of the traced RPC within
+  // the current chunk that is being sent.
+  // - num_traced_bytes_in_chunk: Number of bytes belonging to the traced RPC
+  // within the current chunk.
+  static void Append(ContextList** tail, void* context,
+                     size_t byte_offset_in_stream,
+                     int64_t relative_start_pos_in_chunk,
+                     int64_t num_traced_bytes_in_chunk);
 
-  // Executes a function \a fn with each context in the list and \a ts. It also
-  // frees up the entire list after this operation. It is intended as a callback
-  // and hence does not take a ref on \a error
+  // Executes the function set using grpc_http2_set_write_timestamps_callback
+  // method with each context in the list and \a ts. It also frees up the entire
+  // list after this operation. It is intended as a callback and hence does not
+  // take a ref on \a error. The fn receives individual contexts in the same
+  // order in which they were Appended.
   static void Execute(void* arg, Timestamps* ts, grpc_error_handle error);
 
+  // Executes the passed function \a cb with each context in the list. The
+  // arguments provided to cb include the trace_context_, byte_offset_,
+  // traced_bytes_relative_start_pos_ and num_traced_bytes_ for each context in
+  // the context list. It also frees up the entire list after this operation.
+  // The cb receives individual contexts in the same order in which they were
+  // Appended.
+  static void ForEachExecuteCallback(
+      ContextList* tail,
+      std::function<void(void*, size_t, int64_t, int64_t)> cb);
```

Comment:
Prefer `absl::FunctionRef` - we don't need to hold this function longer

Reply:
Fixed